Source/Core/Core/State.cpp

```diff
@@ -354,6 +337,25 @@ static std::map<double, int> GetSavedStates()
   return m;
 }
 
+static int CompressBufferToFile(const u8* raw_buffer, u32 max_bytes_to_compress, File::IOFile& f)
```

Comment:
I was kinda hoping you'd include the entire loop in the helper function, so the user doesn't have to care about that and can just compress arbitrarily-sized data with a single call.

Reply:
Sure I can do that no problem!